src/components/PermissionsTable/PermissionsTable.tsx

```diff
@@ -0,0 +1,193 @@
+            <TableCell>Domain</TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {(fields || []).map((item, index) =>
+          (<TableRow
+            key={item.id}
+            sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
+          >
+            <TableCell
+            >
+              <IconButton
+                color="primary"
+                onClick={() => remove(index)}
+              >
+                <RemoveIcon />
+              </IconButton>
+            </TableCell>
+            <TableCell>
+              <Typography key={`permissions.${index}.type`}>{item.type}</Typography>
+            </TableCell>
+            <TableCell>
+              <Typography key={`permissions.${index}.layer`}>{item.layer}</Typography>
+            </TableCell>
+            <TableCell>
+              <Typography key={`permissions.${index}.sensitivity`}>{item.sensitivity}</Typography>
+            </TableCell>
+            <TableCell>
+              <Typography key={`permissions.${index}.domain`}>{item.domain}</Typography>
+            </TableCell>
+          </TableRow>)
+          )}
+          <TableRow>
+            <TableCell>
+              <IconButton
+                color="primary"
+                onClick={async () => {
+                  const result = await triggerPermission(undefined, { shouldFocus: true });
+                  if (result) {
+                    const permissionToAdd = watchPermission()
+                    append(permissionToAdd)
+                    resetPermission({
+                      type: undefined,
+                      layer: undefined,
+                      sensitivity: undefined,
+                      domain: undefined,
+                    })
+                  }
+                }}
+              >
+                <AddIcon />
+              </IconButton>
+            </TableCell>
+            <TableCell>
+              <Controller
+                name={'type'}
+                control={controlPermission}
+                render={({ field, fieldState: { error } }) => (
+                  <Select
+                    {...field}
+                    error={!!error}
+                    value={field.value ? field.value : ''}
+                    helperText={error?.message}
+                    native
+                    inputProps={{
+                      'data-testid': `field-type`
+                    }}
+                  >
+                    <option value={''}>Action</option>
+                    {generateUniquePermissionOptions('type')}
+                  </Select>
+                )}
+              />
+            </TableCell>
+            <TableCell>
+              <Controller
+                name={'layer'}
+                control={controlPermission}
+                render={({ field, fieldState: { error } }) => (
+                  dataActions.includes(watchPermission(`type`)) &&
+                  <Select
```

Comment:
I think these selects can be broken out into their own function where we can just pass the relevant information in as parameters, could even go one up to include the controller aswell but not too fussed about that

Reply:
I think this comment still stands